src/RageSoundPosMap.cpp

```diff
@@ -133,38 +119,35 @@ std::int64_t pos_map_queue::Search( std::int64_t iSourceFrame, bool *bApproximat
 		return 0;
 	}
 
-	/* iSourceFrame is probably in pos_map.  Search to figure out what position
-	 * it maps to. */
+	// iSourceFrame is probably in pos_map.  Search to figure out what position it maps to.
 	std::int64_t iClosestPosition = 0, iClosestPositionDist = INT_MAX;
```

Comment:
is `INT_MAX (signed int max, 32-bit at the moment on desktop windows/Linux)` the correct value to start on here? Rather than `std::numeric_limits<int64_t>::max() (explicitly signed 64-bit int max)`?

I assume this just needs to be higher than any possible distance, but if everything else is int64 it'd be more 'correct'.

Reply:
I'll test to see what the compiler is doing, and switch to `std::numeric_limits<int64_t>::max()` if `INT_MAX` isn't actually maxing out a 64 bit integer (and it doesn't negatively impact anything timing related) 🙂

Reply:
Indeed, neither gcc or msvc was maxing out the int64_t when INT_MAX is called. Also, msvc is doing some additional weirdness here that may be slow. I may simply define `iClosestPositionDist` as a constant, in that case.

https://godbolt.org/z/WaTrWxMK8

I had good sync doing a few test plays just now with the `iClosestPositionDist` set to the maximum size of an `int64_t`, and didn't have any of the `frame is out of the range` errors logged except for where it's expected (when you get back to the song wheel after playing a song).

Reply:
You didn't turn on optimizations in your comparison. Any sane compiler will inline the value for `std::numeric_limits<int64_t>::max()`

Reply:
> You didn't turn on optimizations in your comparison. Any sane compiler will inline the value for `std::numeric_limits<int64_t>::max()`

Thank you for noting this. I already made a commit removing the `climits` dependency and put the value in, but I'm happy to return `climits` and use `std::numeric_limits<int64_t>::max()` for `iClosestPositionDist` if the project maintainers prefer that.

https://github.com/itgmania/itgmania/blob/067658c4968aab435f2f6b82cc16fb8116947f49/src/RageSoundPosMap.cpp#L122